tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaMessageOrderTestBase.java

```diff
@@ -162,18 +162,21 @@ public void testKafkaProduceMessageOrder(int batchSize) throws Exception {
             @Cleanup
             KConsumer kConsumer = new KConsumer(topicName, getKafkaBrokerPort(), "testKafkaProduce-KafkaConsume");
             kConsumer.getConsumer().subscribe(Collections.singleton(topicName));
+            int[] receivedKeys = new int[totalMsgs];
             for (int i = 0; i < totalMsgs; ) {
                 ConsumerRecords<Integer, String> records = kConsumer.getConsumer().poll(Duration.ofSeconds(1));
-                if (log.isDebugEnabled()) {
-                    for (ConsumerRecord<Integer, String> record : records) {
-                        log.debug("Kafka consumer get i: {} message: {}, key: {}", i, record.value(), record.key());
-                        assertEquals(record.key().intValue(), i);
-                        i++;
+                for (ConsumerRecord<Integer, String> record : records) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Kafka consumer get i: {} offset: {}, message: {}, key: {}",
+                                i, record.offset(), record.value(), record.key());
                     }
-                } else {
-                    i += records.count();
+                    receivedKeys[i++] = record.key();
```

Comment:
If the message lost, the loop would never stop and the test would fail with timeout so that we cannot see the received keys. I think we can add a timeout for the whole for loop. A simple way is to increase the poll timeout (to 5 seconds for example) and break the loop if `records` is empty.

Reply:
Good point.

Reply:
Actually, we can use the debug log to see the received keys, if the message lost.